### PR TITLE
user registration by admin

### PR DESCRIPTION
**Pull request by nodeexcel**

### CODE REVIEW

File: src/routes/auth.js [hunk 1]
```diff
@@ -6,10 +6,10 @@ export default (app) => {
     app.route("/verify").get(auth.verifyToken);
 
     /* Route for User Registration  */
-    app.route("/user/register").post(user.create);
+    app.route("/user/register").post(auth.requiresAdmin, user.create);
```
Comment (upexcel) [inline src/routes/auth.js]: rename this to create User or add user

File: src/controllers/user.js [hunk 2]
```diff
@@ -8,19 +8,21 @@ export class UserController extends BaseAPIController {
     create = (req, res) => {
         UserProvider.create(this._db.User, req.checkBody, req.body, req.getValidationResult())
             .then((user) => {
-                if (user.user_type == constant().userType.admin || user.user_type == constant().userType.hr || user.user_type == constant().userType.guest) {
+                if ((user.user_type == constant().userType.admin || user.user_type == constant().userType.hr || user.user_type == constant().userType.guest) && (req.user.id)) {
```
Comment (upexcel) [inline src/controllers/user.js]: use a variable instead of repeating user.user_type

File: src/controllers/user.js [hunk 3]
```diff
@@ -8,19 +8,23 @@ export class UserController extends BaseAPIController {
     create = (req, res) => {
         UserProvider.create(this._db.User, req.checkBody, req.body, req.getValidationResult())
             .then((user) => {
-                if (user.user_type == constant().userType.admin || user.user_type == constant().userType.hr || user.user_type == constant().userType.guest) {
+                let user_type = user.user_type;
+                let allowed_role = constant().userType;
+                if ((user_type == allowed_role.admin || user_type == allowed_role.hr || user_type == allowed_role.guest) && (req.user.id)) {
                     this._db.User.create(user)
                         .then((data) => {
                             res.json({
```
Comment (nodeexcel) [inline src/controllers/user.js]: where is condition to check email already exists????

File: src/controllers/user.js [hunk 4]
```diff
@@ -8,19 +8,23 @@ export class UserController extends BaseAPIController {
...
                                 data
```
Comment (nodeexcel) [inline src/controllers/user.js]: user proper response
{
error: 0/1,
message : '',
data : 
}

File: src/controllers/user.js (hunk 3 above)
Comment (nodeexcel) [inline src/controllers/user.js]: for this hooks  is created in model

File: src/controllers/user.js [hunk 5]
```diff
@@ -8,19 +8,29 @@ export class UserController extends BaseAPIController {
...
             .then((user) => {
-                if (user.user_type == constant().userType.admin || user.user_type == constant().userType.hr || user.user_type == constant().userType.guest) {
+                let user_type = user.user_type;
+                let allowed_role = constant().userType;
+                if ((user_type == allowed_role.admin || user_type == allowed_role.hr || user_type == allowed_role.guest) && (req.user.id)) {
                     this._db.User.create(user)
                         .then((data) => {
                             res.json({
-                                data
+                                status: 1,
```
Comment (nodeexcel) [inline src/controllers/user.js]: error : 0

**Pull request merged**